elastic/kibana
# [Discover] Fix unmapped field document explorer rendering

## Summary

This PR fixes rendering of unmapped fields with brackets when using the document explorer and fetching fields from `_source`

Fixes #122665

### Testing

1. Create an index with non-mapped field
```
PUT doc-explorer

PUT doc-explorer/_mapping
{
  "dynamic": false
}

PUT doc-explorer/_doc/1
{
  "a": "a"
}
```

2. Create an index pattern "doc-explorer"
3. Set doc_table:legacy: Off in Kibana advanced settings
4. Open Discover
5. Select doc-explorer data view
6. Add field "a" as a column in the table
7. Observe that the value "a" of field a is displayed without double quotes and brackets, like this:

![image](https://user-images.githubusercontent.com/463851/149802383-8ce6b94b-822d-4c1b-875c-d07a3a0d5849.png)

### Checklist
